docs/web/usecases-components-commands-operations.mdx

```diff
@@ -0,0 +1,123 @@
+
+## Running Commands and Operations
+
+You can run command and operations [through the `UIContext` of the component](sdk-ui-context.mdx).
+
+### Example: Display a Confirmation Dialog to a User
+
+The following example executes the `ui.confirm` operation through the UIContext. Calling a command or operation is as simple as retrieving it by name, providing the appropiate generic type for the command or operation, and then calling `execute` with the appropiate arguments.
+
+:::note
+In order to pass arguments to a command or operation, you have to use the correct type for the generic `command` or `operation` function call with the argument and output type for the command or operation. For example, `ui.confirm` takes in a `ConfirmOperationArgs` and returns a `boolean`, as seen in the [API docs](api-commands-operations.mdx#uiconfirm).
+:::
+
+<Tabs
+  defaultValue="component"
+  values={[
+    { label: 'Component', value: 'component', },
+    { label: 'UI', value: 'ui', },
+  ]
+}>
+<TabItem value="component">
+
+```tsx title="src/components/CustomComponent/CustomComponent.tsx"
+import React, { useContext } from "react";
+import { LayoutElement } from "@geocortex/web/components";
+import { UIContext } from "@geocortex/web/ui";
+import Button from "@geocortex/web/ui/button/Button";
+import { ConfirmOperationArgs } from "@geocortex/viewer-framework/messaging/registry/ui";
+
+export default function CustomComponent(props) {
+    const { operation, command } = useContext(UIContext);
+
+    const displayConfirmDialog = async () => {
+        const result = await operation<ConfirmOperationArgs, boolean>(
+            "ui.confirm"
+        ).execute({
+            title: "Confirm Me",
+        });
+
+        if (result) {
+            console.log("I have been confirmed!");
+        } else {
+            console.warn("I was not confirmed.");
+        }
+    };
+    return (
+        <LayoutElement {...props}>
+            <Button onClick={displayConfirmDialog}>
+                Display Confirm Dialog
+            </Button>
+        </LayoutElement>
+    );
+}
+```
+
+</TabItem>
+<TabItem value="ui">
+<img src={useBaseUrl("img/web-display-notification.png")}/>
+</TabItem>
+</Tabs>
+
+## Implementing Commands and Operations
+
+Components can also register an implementation for any command. This allows for multiple actions to occur on a command execution, and a tighter coupling of command implementation to the UI.
+
+### Example: Component that Implements a Command
+
+When registering a command implementation for a component, it's best practice to register the implementation in the [components model](concepts-components-services.mdx#component-models-and-configuration). The following component model registers an implementation for a custom command that takes a `string` argument.
```

Comment:
Should we be giving example code / referencing to registering commands/operations in a component at all then for this first pass? If we decide to leave the example code, I'd want to leave the first paragraph to give the example context, and if we remove the tip, we should probably remove the content corresponding to commands and operation in the `UIContext` article.

cc: @ianschmitz

Reply:
Everything you wrote about implementing in the component model makes sense for 5.6. It'll still be true in 5.7, but the syntax will differ (using decorators). However since implementing in React components will be *impossible*, I was thinking that part should be removed. So change the sentence to "When registering a command implementation for a component, register the implementation in the components model" (should be component's). And then remove the tip. I think the rest is OK, even though it will change?

Reply:
Yes. that sounds perfect. Changed as suggested.